zproject/backends.py

```diff
@@ -541,6 +545,19 @@ def email_belongs_to_ldap(realm: Realm, email: str) -> bool:
     else:
         return False
 
+def ensure_user_groups_for_ldap_sync_exist(realm):
```

Comment:
```suggestion
def ensure_user_groups_for_ldap_sync_exist(realm: Realm) -> None:
```
needs type annotation

Reply:
we'll end up changing it, but since this is in the single-realm model right now, the function should sanity check it's called correctly I'd say with `assert realm.id == settings.LDAP_GROUP_SYNC_REALM_ID` I'd say

Reply:
I think I'll directly go to per-realm config. That would eliminate all config Parameters except `LDAP_GROUP_SYNC_LIST`.

Format is then:

```python
LDAP_GROUP_SYNC_LIST = {
    <realm_id> : [
        "group1",
        "group2"
    ]
}
```